api/query_test.go

```diff
@@ -395,24 +392,35 @@ func TestFenceHTTPResponse(t *testing.T) {
 	}
 }
 
-func assertError(w *httptest.ResponseRecorder, t *testing.T,
-	testcase struct {
-		name  string
-		error string
-	}) {
-	testErrorInfo := &TestErrorInfo{}
+func setupMock(r *gin.Engine) {
```

Comment:
setupMockServer? or setupTestServer, as it's not mocking, its running production code in a test environment